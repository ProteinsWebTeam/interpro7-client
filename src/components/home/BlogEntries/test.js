import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { BlogEntry, BlogEntries } from '.';

const renderer = new ShallowRenderer();

describe('<BlogEntry />', () => {
  test('Blog entry', () => {
    renderer.render(
      <BlogEntry
        category={'interpro'}
        excerpt={'About InterPro'}
        title={'Protein Families classification'}
        url={
          'https://proteinswebteam.github.io/interpro-blog/2017/10/03/Homologous-superfamily/'
        }
      />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

describe('<BlogEntries />', () => {
  test('Blog entries', () => {
    renderer.render(
      <BlogEntries
        data={{
          loading: false,
          payload: [
            {
              category: 'interpro',
              excerpt: 'About InterPro',
              title: 'Protein Families classification',
              url:
                'https://proteinswebteam.github.io/interpro-blog/2017/10/03/Homologous-superfamily/',
            },
          ],
        }}
      />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
