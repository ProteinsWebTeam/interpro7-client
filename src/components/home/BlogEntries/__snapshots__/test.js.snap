// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogEntries /> Blog entries 1`] = `
<section>
  <ResizeObserverComponent
    element="div"
    measurements={
      Array [
        "width",
      ]
    }
  >
    [Function]
  </ResizeObserverComponent>
  <div
    className="flex-column read-all"
  >
    <Memo(Connect(Link))
      href="https://proteinswebteam.github.io/interpro-blog"
      target="_blank"
    >
      <div
        className="button margin-bottom-none"
      >
        Read all articles
      </div>
    </Memo(Connect(Link))>
  </div>
</section>
`;

exports[`<BlogEntry /> Blog entry 1`] = `
<div
  className="flex-card"
  data-testid="blog-entries-box"
>
  <div
    className="card-image image-blog-biological"
  >
    <div
      className="card-tag tag-blog"
    >
      <Tooltip
        title="View all Blog articles"
      >
        <Memo(Connect(Link))
          className="white-link"
          href="https://proteinswebteam.github.io/interpro-blog/categories/interpro/"
          target="_blank"
        >
          Blog
        </Memo(Connect(Link))>
      </Tooltip>
    </div>
  </div>
  <div
    className="card-content"
  >
    <div
      className="card-title"
    >
      <h4>
        <Memo(Connect(Link))
          href="https://proteinswebteam.github.io/interpro-blog/2017/10/03/Homologous-superfamily/"
          target="_blank"
        >
          Protein Families classification
        </Memo(Connect(Link))>
      </h4>
    </div>
    <div
      className="card-info"
    />
    <div
      className="card-description"
      dangerouslySetInnerHTML={
        Object {
          "__html": "About InterPro",
        }
      }
    />
  </div>
  <Memo(Connect(Link))
    className="card-more"
    href="https://proteinswebteam.github.io/interpro-blog/2017/10/03/Homologous-superfamily/"
    target="_blank"
  >
    <div
      className="button-more icon icon-common icon-right"
      data-icon=""
    >
      Read more
    </div>
  </Memo(Connect(Link))>
</div>
`;
